### ISSUE TITLE
fails to set session_token and cookie at the same time when running in monitoring mode

### ISSUE THREAD

**Issue by mkody** [frozenpandaman/splatnet2statink#9]
Splatfest only starts in 2 hours for me, weird that it passed `mode == 'fest'`

```
Traceback (most recent call last):
  File "splatnet2statink.py", line 685, in <module>
    post_battle(i, results, is_s, is_t, is_m, debug)
  File "splatnet2statink.py", line 560, in post_battle
    title_before = results[i]["player_result"]["player"]["fes_grade"]["rank"]
KeyError: 'fes_grade'
```

**Comment by frozenpandaman** [comment]
Found the culprit…

https://github.com/frozenpandaman/splatnet2statink/commit/058876166638d8f02434c147b3efc4e8f437b251#diff-2215b58f10f3ed4fb0e9e35a03ccf764R474

Fixing. Thanks. :)

**Comment by mkody** [comment]
Thanks!
That commit as also introduced this error now when using the monitor mode:
```
Traceback (most recent call last):
  File "splatnet2statink.py", line 680, in <module>
    monitor_battles(is_s, is_t, debug)
  File "splatnet2statink.py", line 167, in monitor_battles
    for result in results:
UnboundLocalError: local variable 'results' referenced before assignment
```

**Comment by frozenpandaman** [comment]
@mkody,

I got this notification in my email, but it appears you've deleted the comment as I don't see it here:
![image](https://user-images.githubusercontent.com/7245874/30001889-7d10a882-9050-11e7-8b07-551480a045e0.png)

Are you still having this issue? Monitoring mode's working fine for me, but please try downloading the latest version and do let me know if you have any issues.

**Comment by mkody** [comment]
I got this but I couldn't reproduce it later.

This is how it went: 

- I did a `git pull`
- Start `python2 splatnet2statink.py -M`
- It asks for my stat.ink key, I give it (huh, did `config.txt` get overwritten to blank again?)
- Asks me to login, I give the "Use this account" link back
- And then this error happen
- Starting again `python2 splatnet2statink.py -M`
- UnboundLocalError again
- Start `python2 splatnet2statink.py`, it works and I can upload the results
- Try once again `python2 splatnet2statink.py -M`, no more errors and it works

**Comment by frozenpandaman** [comment]
I was able to reproduce it. Interesting…

Using a blank config.txt, immediately running it with `-M` seems to return the error immediately after printing out "Wrote session_token to config.txt." So, the cookie is not being set. Running the exact same command again generates and sets the cookie in config.txt ("Wrote iksm_session cookie to config.txt.") but then immediately throws the error again. Running a third time, it works fine. So, something weird about the way the file is being written to/read from, I think. Will look into it tomorrow. Thanks!

**Comment by frozenpandaman** [comment]
Hm, something weird on Nintendo's end – `load_json()` is returning the wrong data, since accessing https://app.splatoon2.nintendo.net/api/results immediately after setting the session_token and cookie gives a `{'code': 'INTERNAL_SERVER_ERROR'}` (instead of displaying the json data) for some reason.

**Comment by frozenpandaman** [comment]
Should be fixed in 5508aab! It behaves differently when you haven't battled online before (i.e. when I was generating tokens using a test account 😉 ) so that was adding some complications in trying to diagnose the issue…

Anyway, I'm hoping the only time line 175 will fail (and reach the except block) is indeed when you haven't played online before (as the print statement says). If you're getting that error for some reason (and let me know if you are) I have to say I'd be pretty stumped, since I have no clue why `load_json()` would ever return an internal server except for in that case, especially since the session_token and cookie are all without-a-doubt correctly set this time around.
